pkg/engines/jx/controller.go

```diff
@@ -1,570 +1,275 @@
 package jx
 
 import (
-	"encoding/json"
-	"fmt"
+	"context"
 	"os"
-	"os/signal"
-	"reflect"
 	"strconv"
 	"strings"
-	"sync"
-	"syscall"
 	"text/template"
-	"time"
 
-	jsonpatch "github.com/evanphx/json-patch"
 	jxv1 "github.com/jenkins-x/jx-api/pkg/apis/jenkins.io/v1"
-	jxclient "github.com/jenkins-x/jx-api/pkg/client/clientset/versioned"
-	jxinformers "github.com/jenkins-x/jx-api/pkg/client/informers/externalversions/jenkins.io/v1"
-	jxlisters "github.com/jenkins-x/jx-api/pkg/client/listers/jenkins.io/v1"
 	"github.com/jenkins-x/jx/v2/pkg/tekton/metapipeline"
-	"github.com/jenkins-x/lighthouse/pkg/apis/lighthouse/v1alpha1"
-	clientset "github.com/jenkins-x/lighthouse/pkg/client/clientset/versioned"
-	lhinformers "github.com/jenkins-x/lighthouse/pkg/client/informers/externalversions/lighthouse/v1alpha1"
-	lhlisters "github.com/jenkins-x/lighthouse/pkg/client/listers/lighthouse/v1alpha1"
+	lighthousev1alpha1 "github.com/jenkins-x/lighthouse/pkg/apis/lighthouse/v1alpha1"
 	"github.com/jenkins-x/lighthouse/pkg/config"
 	"github.com/jenkins-x/lighthouse/pkg/util"
-	"github.com/jenkins-x/lighthouse/pkg/watcher"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
-	"golang.org/x/time/rate"
-	kubeerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
-	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/tools/cache"
-	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 	"sigs.k8s.io/yaml"
 )
 
 const (
 	controllerName           = "jx-controller"
-	activityKeyPrefix        = "activity"
-	jobKeyPrefix             = "job"
 	defaultTargetURLTemplate = "{{ .BaseURL }}/teams/{{ .Team }}/projects/{{ .Owner }}/{{ .Repository }}/{{ .Branch }}/{{ .Build }}"
 	baseTargetURLEnvVar      = "LIGHTHOUSE_REPORT_URL_BASE"
 	targetURLTeamEnvVar      = "LIGHTHOUSE_REPORT_URL_TEAM"
-	targetURLTemplateEnvVar  = "LIGHTHOUSE_REPORT_URL_TEMPLATE"
+	pipelineActivityKey      = ".metadata.pipelineActivity"
```

Comment:
So I'm confused as to what exactly this is - is it the name of a real field, or is it a field we're creating?

Reply:
It’s the key used by the indexer to lookup resources from the cache

Reply:
Ok, so it's just an arbitrary string?

Reply:
Yes, preferably not something that already exists.

Reply:
Ok, I was quite confused. =)